language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -179,30 +179,38 @@ class DottyLanguageServer extends LanguageServer
     val document = params.getTextDocument
     val uri = new URI(document.getUri)
     val driver = driverFor(uri)
+    val worksheetMode = isWorksheet(uri)
+
+    val (text, positionMapper) =
+      if (worksheetMode) (wrapWorksheet(document.getText), worksheetPositionMapper _)
+      else (document.getText, identity[SourcePosition] _)
```

Comment:
I would rather make this parameter an Option[Position => Position] so we can pass None here.